Data Preparation.R

```diff
@@ -0,0 +1,173 @@
+                      "inpatient_daycase_identifier from SMR01_PI ",
+                      "where discharge_date >= {d '", start_date, "'} 
+                       AND discharge_date <= {d '", end_date, "'} ",
+                      "ORDER BY link_no, cis_marker, admission_date, 
+                       discharge_date, admission, discharge, uri")
+
+data_01 <- as_tibble(dbGetQuery(SMRA, Query_SMR01)) %>%
+           mutate(recid = "SMR01") %>% clean_names()
+
+
+## 2. SMR50 extract ##
+Query_SMR50 <- paste0("select link_no, gls_cis_marker, admission_date, 
+                       discharge_date, sex, ",
+                      "inpatient_daycase_identifier from ANALYSIS.SMR01_1E_PI ",
+                      "where discharge_date >= {d '", start_date, "'} 
+                       AND discharge_date <= {d '", end_date, "'} ",
+                      "ORDER BY link_no, cis_marker, admission_date, 
+                       discharge_date, admission, discharge, uri")
+
+data_50  <- as_tibble(dbGetQuery(SMRA, Query_SMR50)) %>%
+            mutate(recid = "SMR50") %>% clean_names()
+
+
+## 3. SMR04 extract ##
+Query_SMR04 <- paste("select link_no, cis_marker, admission_date, 
+                      discharge_date, sex,",
+                     "management_of_patient from ANALYSIS.SMR04_PI ",
+                     "WHERE management_of_patient IN ('1', '3', '5', '7', 'A')",
+                     "ORDER BY link_no, cis_marker, admission_date, 
+                      discharge_date, admission, discharge, uri")
+
+data_04 <- as_tibble(dbGetQuery(SMRA, Query_SMR04)) 
+data_04 %<>% filter(DISCHARGE_DATE >= "2009-10-01" & 
+                    DISCHARGE_DATE < "2018-03-31") %>% drop_na() %>%
+                    mutate(recid = "SMR04") %>% clean_names()
+
+
+## 4. Deaths extract ##
+Query_Deaths <- paste0("select link_no, date_of_death, institution, sex, ",
+                       "underlying_cause_of_death, cause_of_death_code_0, ",
+                       "cause_of_death_code_1, cause_of_death_code_2, ",
+                       "cause_of_death_code_3, cause_of_death_code_4, ",
+                       "cause_of_death_code_5, cause_of_death_code_6, ",
+                       "cause_of_death_code_7, cause_of_death_code_8, ",
+                       "cause_of_death_code_9 ",
+                       "from ANALYSIS.GRO_DEATHS_C ",
+                       "where date_of_death >= {d '", deaths_start_date, "'}")
+
+data_deaths  <- as_tibble(dbGetQuery(SMRA, Query_Deaths)) %>%
+                mutate(recid = "Deaths") %>% clean_names()
+
+
+#######################################
+## SECTION 3 - Data preparation      ##
+#######################################
+
+
+# 1. Create SMR activity file (SMR01_50)
+
+smr01_50 <- rbind(data_01,data_50) %>% 
+  
+# 2. Select only inpatient_day case
+  
+filter(inpatient_daycase_identifier == "I") %>% 
+  
+# 3. Aggregate to CIS level. Record first admission and last discharge dates.
+  
+group_by(link_no,gls_cis_marker) %>% 
```

Comment:
Formatting of this section could be improved; e.g. 
```suggestion
smr01_50 %<>%
   group_by(link_no,gls_cis_marker) %>% 
   summarise(admission_date = min(admission_date),
             discharge_date = max(discharge_date)) %>%
   ungroup()
```